src/OpenFeature.Contrib.Providers.Flagd/InProcessResolver.cs

```diff
@@ -0,0 +1,276 @@
+using System.Net.Http;
+using System.Threading.Tasks;
+using OpenFeature.Model;
+using OpenFeature.Flagd.Grpc.Sync;
+using System;
+using System.IO;
+using System.Linq;
+using System.Net.Security;
+using System.Security.Cryptography.X509Certificates;
+using Grpc.Net.Client;
+using OpenFeature.Flagd.Grpc;
+using System.Net.Sockets;
+using System.Threading;
+using Grpc.Core;
+using Value = OpenFeature.Model.Value;
+
+namespace OpenFeature.Contrib.Providers.Flagd
+{
+    internal class InProcessResolver : Resolver
+    {
+        static int EventStreamRetryBaseBackoff = 1;
+
+        private readonly FlagSyncService.FlagSyncServiceClient _client;
+        private readonly JsonEvaluator _evaluator;
+
+        private readonly Mutex _mtx;
+        private int _eventStreamRetries;
+        private int _eventStreamRetryBackoff = EventStreamRetryBaseBackoff;
+        private readonly FlagdConfig _config;
+
+        internal InProcessResolver(FlagdConfig config)
+        {
+            _config = config;
+            _client = BuildClient(config, channel => new FlagSyncService.FlagSyncServiceClient(channel));
+            _mtx = new Mutex();
+            _evaluator = new JsonEvaluator(config.SourceSelector);
+        }
+
+        internal InProcessResolver(FlagSyncService.FlagSyncServiceClient client, FlagdConfig config)
+        {
+            _config = config;
+            _client = client;
+            _mtx = new Mutex();
+            _evaluator = new JsonEvaluator(config.SourceSelector);
+        }
+
+
+        public void Init()
+        {
+            var handleEvents = new Thread(HandleEvents);
+            handleEvents.Start();
+        }
```

Comment:
This method needs to block until the connection is established, or else the setProvider returns and the READY event fires before the provider is actually READY (and indeed in might never bet ready). I would use some kind of latch or callback or something to block until we connect.

As a hacky fix you can add a Thread.sleep here for 1000 and things "work".